src/toposort.rs

```diff
@@ -193,60 +196,73 @@ impl TopologicalSorter {
     ///     of the nodes given to :meth:`done`.  This can only happen if the ``initial`` nodes had
     ///     even a partial topological ordering amongst themselves, which is not a valid
     ///     starting input.
-    fn done(&mut self, py: Python, nodes: Vec<usize>) -> PyResult<()> {
-        let dag = &self.dag.borrow(py);
-        let (in_dir, out_dir) = traversal_directions(self.reverse);
-        for node in nodes {
-            let node = NodeIndex::new(node);
-            match self.node2state.get_mut(&node) {
-                None => {
-                    return Err(PyValueError::new_err(format!(
-                        "node {} was not passed out (still not ready).",
-                        node.index()
-                    )));
-                }
-                Some(NodeState::Done) => {
-                    return Err(PyValueError::new_err(format!(
-                        "node {} was already marked done.",
-                        node.index()
-                    )));
-                }
-                Some(state) => {
-                    debug_assert_eq!(*state, NodeState::Ready);
-                    *state = NodeState::Done;
-                }
+    fn done(&mut self, nodes: &Bound<PyAny>) -> PyResult<()> {
+        if let Ok(node) = nodes.extract::<usize>() {
+            self.done_single(nodes.py(), NodeIndex::new(node))
+        } else if let Ok(nodes) = nodes.downcast::<PyList>() {
+            for node in nodes {
+                self.done_single(nodes.py(), NodeIndex::new(node.extract()?))?
+            }
+            Ok(())
+        } else {
+            for node in nodes.iter()? {
+                self.done_single(nodes.py(), NodeIndex::new(node?.extract()?))?
+            }
+            Ok(())
+        }
+    }
+}
+
+impl TopologicalSorter {
+    #[inline(always)]
+    fn done_single(&mut self, py: Python, node: NodeIndex) -> PyResult<()> {
+        let dag = self.dag.borrow(py);
+        match self.node2state.get_mut(&node) {
+            None => {
+                return Err(PyValueError::new_err(format!(
+                    "node {} was not passed out (still not ready).",
+                    node.index()
+                )));
+            }
+            Some(NodeState::Done) => {
+                return Err(PyValueError::new_err(format!(
+                    "node {} was already marked done.",
+                    node.index()
+                )));
+            }
+            Some(state) => {
+                debug_assert_eq!(*state, NodeState::Ready);
```

Comment:
Did you intend to leave this here?

Reply:
I didn't add it - it was just there in the original. It doesn't hurt in release mode (it gets compiled out), and it expands the test coverage a bit in debug.

Reply:
It's fine to leave it. I know it doesn't affect the release, but we rarely use the `debug_assert` in rustworkx.